include/seqan3/alphabet/union_alphabet.hpp

```diff
@@ -129,13 +196,48 @@ struct union_alphabet
     //! the type of the alphabet when converted to char (e.g. via @link to_char @endlink)
     using char_type = typename first_alphabet_type::char_type;
     //! the type of the alphabet when represented as a number (e.g. via @link to_integral @endlink)
-    using integral_type = decltype(detail::at_least_type<value_size>());
+    using integral_type = detail::at_least_type_t<value_size>;
+    /*! the type used to assign a value from one of the base alphabets during
+     * copy construction or copy assignment. (i.e. via
+     * @link union_alphabet(const variant_type & alphabet) @endlink or
+     * @link operator= (const variant_type & alphabet) @endlink )
+     */
+    using variant_type = std::variant<first_alphabet_type, alphabet_types...>;
+
+    //! @name default constructors
+    //!@{
+    constexpr union_alphabet() = default;
+    constexpr union_alphabet(const union_alphabet &) = default;
+    constexpr union_alphabet(union_alphabet &&) = default;
+    constexpr union_alphabet(const integral_type & value) : value{value}{}
+    //!@}
 
-protected:
-    using tuple_t = std::tuple<first_alphabet_type, alphabet_types...>;
-    using index_sequence_t = decltype(std::make_index_sequence<std::tuple_size_v<tuple_t>>{});
-    static constexpr auto sizes = detail::alphabet_sizes<tuple_t>(index_sequence_t{});
-    static constexpr auto alphabets = detail::alphabet_objects<tuple_t>(index_sequence_t{});
+    //! @name default assignment operators
+    //!@{
+    constexpr union_alphabet & operator= (const union_alphabet &) = default;
+    constexpr union_alphabet & operator= (union_alphabet &&) = default;
+    constexpr union_alphabet(integral_type && value) : value{value}{}
+    //!@}
+
+    /*! allow construction via a value of the base alphabets
+     *~~~~~~~~~~~~~~~{.cpp}
+     *     union_alphabet<dna4, gap> letter1{dna4::C}; // or
+     *     union_alphabet<dna4, gap> letter2 = gap::GAP;
+     *~~~~~~~~~~~~~~~
+     */
+    constexpr union_alphabet(const variant_type & alphabet) : value(from_base_(alphabet)){}
```

Comment:
style guide + readability:
```c++
constexpr union_alphabet(variant_type const & alphabet) : 
    value{from_base_(alphabet)}
{}
```
(in any case the initialization of value should happen with braces and it should be obvious that the empty braces are a function definition and not related to value)